packages/core/src/common/abstractComponent.ts

```diff
@@ -42,9 +42,9 @@ export abstract class AbstractComponent<P, S> extends React.Component<P, S> {
      * @returns a "cancel" function that will clear timeout when invoked.
      */
     public setTimeout(callback: () => void, timeout?: number) {
-        const handle = setTimeout(callback, timeout);
+        const handle = window.setTimeout(callback, timeout);
         this.timeoutIds.push(handle);
-        return () => clearTimeout(handle);
+        return () => window.clearTimeout(handle);
```

Comment:
curious: why add the `window` reference?

Reply:
with dependency hoisting, we get `@types/node` at the root. the global `setTimeout` resolves to node's `setTimeout` before it sees `window.setTimeout` in lib.d.ts. so we make this more explicit

Reply:
would be cool to add a code comment to that effect

Reply:
it's pretty obvious when you make the mistake. it's a compile error. the typedef takes you to a different file. not going to add a comment

Reply:
what do you think about tslint `ban`ning non-window references to `set/clearTimeout/Interval`?